apps/dashboard/src/hooks/use-sync-workflow.tsx

```diff
@@ -63,7 +63,18 @@ export function useSyncWorkflow(workflow: WorkflowListResponseDto) {
     return showToast({
       variant: 'lg',
       className: 'gap-3',
-      children: ({ close }) => <PromoteSuccessToast workflow={workflow} environment={environment} onClose={close} />,
+      children: ({ close }) => (
+        <SuccessButtonToast
+          title={`Workflow synced to ${environment?.name}`}
+          description={`Workflow '${workflow.name}' has been successfully synced to ${environment?.name}.`}
+          actionLabel={`Switch to ${environment?.name}`}
+          onAction={() => {
+            close();
+            switchEnvironment(environment?.slug || '');
+          }}
+          onClose={close}
+        />
+      ),
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** The `environment` should be defined at this point. If it's easy to do so, can we fix the types or return a loading state whilst it's not defined?‏

Reply:
This is an issue across the new Dashboard as the returned type of `environment` from the `useEnvironment` hook. We can use discriminated unions to make sure the environment is not optional if a loaded: true flag is true or we can use an env guard that ensures all its children have an environment set. 

I suggest we tackled this in a separate PR across all the codebase.